packages/ecmascript/src/Shims.ts

```diff
@@ -124,30 +124,41 @@ export default class Shims {
 			case "Map":
 				arrayLikeClone = (arrayLike as Map<any, any>).entries();
 				break;
-			default: return arrayLikeClone;
-		}
-
-		if (mapfn) {
-			array = arrayLikeClone.map(mapfn);
-		} else {
-			array = arrayLikeClone;
+			case "Object":
+				arrayLikeClone = [];
+				// Check if the object is an array-like object
+				if (Object.keys(arrayLike).find(item => item.indexOf('length') >= 0)) {
+					const length = (arrayLike as ArrayLike<any>).length;
+					// mimic the behavior of the standard Array.from() method
+					arrayLikeClone = Array.apply(null, Array(length));// nosonar
+					Object.keys(arrayLike).forEach(element => {
+						const indexKey = parseInt(element);
+						// Check if object key is like an indexed element and within range
+						if (!isNaN(indexKey) && indexKey < length) {
+							arrayLikeClone[element] = arrayLike[element];
+						}
+					});
+				}
+				break;
+			default:
+				return arrayLikeClone;
```

Comment:
Oh something that came to mind later. Maybe default to returning an empty array? 
If the shim can't handle it, it should still honor its return type (which is any[])
```suggestion
				return [];
```

Simple example:

```ts
arrayFrom(10) // Output: 10
Array.from(10) // Output: []
```

Reply:
I will take care of it in separate PR

Reply:
@Indy-rbo @akantchev FYI:
https://github.com/vmware/build-tools-for-vmware-aria/pull/372